base/file.jl

```diff
@@ -335,6 +335,8 @@ function cptree(src::AbstractString, dst::AbstractString; force::Bool=false,
         end
     end
 end
+cptree(src::AbstractString, dst::AbstractString; kwargs...) =
+    cptree(String(src)::String, String(dst)::String; kwargs...)
```

Comment:
🤔

Reply:
1. fixes an invalidation in `joinpath` if all it knows is that it's an `::AbstractString`. With the change above, it's guaranteed to be a String.
2. reduces the argument diversity of the "expensive method" (in this case it's not very expensive)
3. the `String(src)::String` prevents a StackOverflowError if someone writes

```julia
String(str::MyFunkyString)= str
```
(which is busted, but there's nothing preventing anyone from doing so)

I wouldn't have changed it if it weren't for the invalidation.

Reply:
> a StackOverflowError if someone writes

I feel like that stack overflow might be a feature, i.e. karma.

Reply:
Only for the guilty party. For anyone else it's really confusing.

Reply:
Sadly true. There is no justice.